Assets/Plugins/Source/Core/Config.cs

```diff
@@ -83,26 +89,76 @@ protected Config(string filename, string directory)
+        /// factory pattern. This requires that each class that derives from Config must implement
+        /// a static method registering it's constructor.
+        /// </summary>
+        /// <typeparam name="T">The config type.</typeparam>
+        /// <param name="factory">The function to create the config type</param>
+        protected static void RegisterFactor<T>(Func<Config> factory) where T : Config
+        {
+            s_factories[typeof(T)] = factory;
+        }
+
+        /// <summary>
+        /// Try to retrieve the factory method for the indicated type that can be used to
+        /// create a new instance of the given config type.
+        /// </summary>
+        /// <typeparam name="T">The type of config to get the factory method for.</typeparam>
+        /// <param name="factory">The factory method that instantiates the config indicated.</param>
+        /// <returns>True if the factory is registered, false otherwise.</returns>
+        /// <exception cref="InvalidOperationException">
+        /// If the indicated type does not have a corresponding constructor function
+        /// registered, a verbose exception will be thrown indicating how to properly
+        /// implement the Config implementing class such that it's constructor is
+        /// properly registered.
+        /// </exception>
+        private static bool TryGetFactory<T>(out Func<Config> factory) where T : Config
+        {
+            if(!s_factories.TryGetValue(typeof(T), out factory))
+            {
+                throw new InvalidOperationException(
+                    $"No factory method has been registered for type \"{typeof(T).FullName}\". " +
+                    $"Please make sure that \"{typeof(T).FullName}\" registers it's constructor with the base Config class via a static constructor.");
+            }
+
+            return true;
+        }
+
         /// <summary>
         /// Retrieves the indicated Config object, reading its values into memory.
         /// </summary>
         /// <typeparam name="T">The Config to retrieve.</typeparam>
         /// <returns>Task<typeparam name="T">Config type.</typeparam></returns>
-        public static async Task<T> GetAsync<T>() where T : Config, new()
+        public static async Task<T> GetAsync<T>() where T : Config
         {
+            // NOTE: This block (and the corresponding one below) exists so that
+            //       the config values are only cached when not in the editor.
+            //       In the editor, config files can be changed, so they should
+            //       not be cached.
 #if !UNITY_EDITOR
+            // Determine if there is a cached copy of the config, return that if it exists.
             if (s_cachedConfigs.TryGetValue(typeof(T), out Config config))
             {
                 return (T)config;
             }
 #endif
+            // Try to get the factory method used to instantiate the config.
+            TryGetFactory<T>(out Func<Config> factory);
```

Comment:
The reason I did not check for this is that the `TryGetFactory` method throws an `InvalidOperationException` if the factory is not found. I did not want to design it in such a way where it would ever be "okay" that `TryGetFactory` returned false - but I wanted to follow the naming convention and pattern that usually accompanies a "`TryGet[Insert Noun Here]`" method in other places of C#.

Reply:
Basically, I wanted to require a `try` block to suppress the lack of a factory, that way it cannot fail silently.